include/nighthawk/adaptive_load/metrics_plugin.h

```diff
@@ -11,6 +11,14 @@
 
 namespace Nighthawk {
 
+// Describes the period in which a Metrics plugin should measure.
```

Comment:
Since this is the public API of the metric plugins, we should expand this doc comment. We need to provide enough information here, so that any developers of metric plugins understand exactly what they need to do.

The current comment just says that this is the period plugin should measure. However we aren't trying to dictate when plugin measures. I think we are trying to tell the plugin over which period should the metric be reported.

With that - maybe `ReportingPeriod` would be a better name?

Bonus points if we also provide an example of how this is used to document it even better.

Reply:
Done.

Reply:
Thank you, the updated doc comment are detailed and easy to understand.